elasticsearch: backend: better timestamp parsing

gabor opened the pull request:
the current main-branch elastic code, when processing metric elastic responses, rounds the numbers to seconds. this should not happen.

the current code uses [time.Unix](https://pkg.go.dev/time#Unix) , i switched to use [time.UnixMIlli](https://pkg.go.dev/time#UnixMilli)

there are two approaches in the pull request:
- the [first commit](https://github.com/grafana/grafana/pull/64681/commits/e90e7d69d24305a8943c8b1233e0d2b5cb346849) shows an approach where we fix the problem by fixing every place where the code is wrong
- if you read the whole pr diff together, it shows an approach where we refactor the code to have one separate `getAsTime` function. unfortunately this is a deeper change, because it needs to be able to handle error-cases.

gabor committed:
elasticsearch: backend: better timestamp parsing

gabor committed:
refactor

ivanahuckova reviewed (approved):
I think that this second commit is better. 🙂

gabor commented:
@ivanahuckova i refactored it more, i have extracted all the switch-cases from `processMetrics` into separate functions, i think this makes them easier to maintain and understand. WDYT?

ivanahuckova reviewed (approved):
LGTM, also really nice refactor! 🤩
